AUTH_PROXY.md

```diff
@@ -0,0 +1,9 @@
+# Authenticating requests behind a proxy
+
+If your instance is behind an authenticating proxy that requires additional headers, they can be supplied via environment variables as follows:
+
+```sh
+SRC_HEADER_NAME=value src search 'foobar'
```

Comment:
I suggest to use real values in examples and use multiple of them.

Why? `SRC_HEADER_NAME=value` can also be read as "the name of the header is set by using `SRC_HEADER_NAME` and the name itself is then the `value`"

If you use `SRC_HEADER_AUTHORIZATION="Bearer my-token"` and explains that this sets `authorization: Bearer my-token` then it's clear what's what.

Reply:
Question: how would you set a header that contains a hyphen in the name, e.g. `Proxy-Authorization`? `SRC_HEADER_PROXY-AUTHORIZATION`? Or is the `-` turned into a `_`?

Reply:
They're copied verbatim, but you'd need to use env to supply them, e.g. `env "SRC_HEADER_FOO-BAR=whyamilikethis" src doathing`